### PR TITLE
[FIX] website_sale: Pricelist witn min qty

### PR DESCRIPTION
**Pull request by simongoffin**
Each time a product is added in the cart, the qty of this product must be checked to
adjust the price if there is a rule (with min qty) in the partner's pricelist.

opw:630049

### CODE REVIEW

**Comment by simongoffin** [comment]
@beledouxdenis

File: addons/website_sale_options/controllers/main.py [hunk 1]
```diff
@@ -61,7 +61,8 @@ def modal(self, product_id, **kw):
             context['pricelist'] = int(pricelist)
 
         website_context = kw.get('kwargs', {}).get('context', {})
-        context = dict(context or {}, **website_context)
+        qty = kw.get('kwargs', {}).get('qty', {})
```
Comment (Gorash) [inline addons/website_sale_options/controllers/main.py]: qty = website_context.get('qty', 1)

File: addons/website_sale_options/controllers/main.py [hunk 2]
```diff
@@ -61,7 +61,8 @@ def modal(self, product_id, **kw):
...
+        context = dict(context or {}, quantity=qty, **website_context)
```
Comment (Gorash) [inline addons/website_sale_options/controllers/main.py]: cancel change

File: addons/website_sale_options/static/src/js/website_sale.js [hunk 3]
```diff
@@ -10,7 +11,8 @@ $(document).ready(function () {
             openerp.jsonRpc("/shop/modal", 'call', {
                     'product_id': product_id,
                     kwargs: {
-                       context: openerp.website.get_context()
+                       context: openerp.website.get_context(),
+                       qty: quantity
```
Comment (Gorash) [inline addons/website_sale_options/static/src/js/website_sale.js]: context: _.extend({'quantity': quantity}, openerp.website.get_context())

File: addons/website_sale_options/static/src/js/website_sale.js [hunk 4]
```diff
@@ -63,6 +65,15 @@ $(document).ready(function () {
                     $('ul.js_add_cart_variants').each(function () {
                         $('input.js_variant_change, select.js_variant_change', this).first().trigger('change');
                     });
+
+                    $modal.on("change", 'input[name="add_qty"]', function (event) {
+                        var qty = $(event.target).closest('tr').find('input[name="add_qty"]').val();
+                        var product_id = $(event.target).closest('tr').find('[data-oe-model="product.product"]').attr('data-oe-id');
```
Comment (Gorash) [inline addons/website_sale_options/static/src/js/website_sale.js]: var product_id = +$('input.product_id:first').val();

File: addons/website_sale/controllers/main.py [hunk 5]
```diff
@@ -933,4 +933,13 @@ def tracking_cart(self, **post):
             ret['lines'] = self.order_lines_2_google_api(order.order_line)
         return ret
 
+    @http.route(['/shop/get_unit_price'], type='json', auth="public", methods=['POST'], website=True)
```
Comment (Gorash) [inline addons/website_sale/controllers/main.py]: need to accept product_ids

File: addons/website_sale/controllers/main.py [hunk 6]
```diff
@@ -933,4 +933,13 @@ def tracking_cart(self, **post):
...
+    def get_unit_price(self, product_id, add_qty, **kw):
```
Comment (Gorash) [inline addons/website_sale/controllers/main.py]: need to accept product_ids

File: addons/website_sale/static/src/js/website_sale.js [hunk 7]
```diff
@@ -19,17 +19,32 @@ $('.oe_website_sale').each(function () {
...
         var $input = $(this);
         var value = parseInt($input.val(), 10);
         var line_id = parseInt($input.data('line-id'),10);
         if (isNaN(value)) value = 0;
-        openerp.jsonRpc("/shop/cart/update_json", 'call', {
+        openerp.jsonRpc("/shop/get_unit_price", 'call', {
+            'product_id': parseInt($input.data('product-id'),10),
+            'add_qty': value})
+        .then(function (res) {
+
```
Comment (Gorash) [inline addons/website_sale/static/src/js/website_sale.js]: change price for all product ids (all variants or options) with same quantity and after trigger("change") on ul[data-attribute_value_ids]

File: addons/website_sale/static/src/js/website_sale.js [hunk 8]
```diff
@@ -19,17 +19,67 @@ $('.oe_website_sale').each(function () {
...
+        });
         if (isNaN(value)) value = 0;
-        openerp.jsonRpc("/shop/cart/update_json", 'call', {
+        openerp.jsonRpc("/shop/get_unit_price", 'call', {
+            'product_ids': product_ids,
+            'add_qty': value})
+        .then(function (res) {
+            //basic case
+            $dom.find('span.oe_currency_value').last().text(res[product_id].toFixed(2));
+            if(res[product_id].toFixed(2)<default_price){
```
Comment (Gorash) [inline addons/website_sale/static/src/js/website_sale.js]: check with at least 1% difference

File: addons/website_sale/static/src/js/website_sale.js [hunk 9]
```diff
@@ -19,17 +19,67 @@ $('.oe_website_sale').each(function () {
...
+        var qty = $(event.target).closest('form').find('input[name="add_qty"]').val();
+
+        openerp.jsonRpc("/shop/get_unit_price", 'call', {'product_ids': product_ids,'add_qty': parseInt(qty)})
+        .then(function (data) {
+            var current = product_dom.data("attribute_value_ids");
+            for(var j=0; j < current.length; j++){
+                current[j][2] = data[current[j][0]];
+            }
+            product_dom.attr("data-attribute_value_ids", JSON.stringify(current)).trigger("change");
+            //$(event.target).closest('form').find(".oe_price .oe_currency_value").text(data.toFixed(2));
```
Comment (Gorash) [inline addons/website_sale/static/src/js/website_sale.js]: remove

File: addons/website_sale/static/src/js/website_sale.js [hunk 10]
```diff
@@ -19,17 +19,67 @@ $('.oe_website_sale').each(function () {
...
+            }
+            else{
+                $dom.find('.text-danger').hide();
+            }
+            //optional case
+            $dom_optional.each(function(){
+                var id = $(this).find('span[data-oe-model="product.product"]').data('oe-id');
+                var price = parseFloat($(this).find(".text-danger > span.oe_currency_value").text());
+                $(this).find("span.oe_currency_value").last().text(res[id].toFixed(2));
+                if(res[id].toFixed(2)<price){
```
Comment (Gorash) [inline addons/website_sale/static/src/js/website_sale.js]: same 1%

File: addons/website_sale_options/static/src/js/website_sale.js [hunk 11]
```diff
@@ -63,6 +63,39 @@ $(document).ready(function () {
...
+                            for(var i=0; i < $products_dom.length; i++) {
+                                current = $products_dom[i].data("attribute_value_ids");
+                                for(var j=0; j < current.length; j++){
+                                    current[j][2] = data[current[j][0]];
+                                }
+                                $products_dom[i].attr("data-attribute_value_ids", JSON.stringify(current)).trigger("change");
+                            }
+                            $dom.find(".oe_price .oe_currency_value").text(data[product_id].toFixed(2));
+                            if(data[product_id].toFixed(2)<default_price){
+                                $dom.find('.text-danger.oe_default_price').show().css('text-decoration', 'line-through');
```
Comment (Gorash) [inline addons/website_sale_options/static/src/js/website_sale.js]: not need text-decoration (because you add it into the template change)

**Pull request closed without merging**